lib/Completion/Bridge/TolerantParser/ReferenceFinder/AttributeCompletor.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Phpactor\Completion\Bridge\TolerantParser\ReferenceFinder;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Phpactor\Completion\Bridge\TolerantParser\CompletionContext;
+use Phpactor\Completion\Bridge\TolerantParser\Qualifier\AttributeQualifier;
+use Phpactor\Completion\Bridge\TolerantParser\TolerantCompletor;
+use Phpactor\Completion\Bridge\TolerantParser\TolerantQualifiable;
+use Phpactor\Completion\Bridge\TolerantParser\TolerantQualifier;
+use Phpactor\Completion\Core\DocumentPrioritizer\DocumentPrioritizer;
+use Phpactor\Completion\Core\Suggestion;
+use Phpactor\ReferenceFinder\NameSearcher;
+use Phpactor\ReferenceFinder\NameSearcherType;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\TextDocument;
+
+class AttributeCompletor implements TolerantCompletor, TolerantQualifiable
+{
+    public function __construct(
+        private NameSearcher $nameSearcher,
+        private DocumentPrioritizer $prioritizer
+    ) {
+    }
+
+    public function complete(Node $node, TextDocument $source, ByteOffset $offset): Generator
+    {
+        if (!CompletionContext::attribute($node)) {
```

Comment:
Is it needed if `AttributeQualifier` checked it before?

Reply:
you could use the "qualifier" or remove it and use this only. I don't think there is any advantage in using the `Qualfiier` (IIRC it was to allow the "qualifiying" logic to be shared, but that's better done with `CompletionContext` anyway).

Reply:
It's nicer for me to work separated pieces than on one bigger class. If it were up to me, I'd leave `AttributeQualifier`. Let me know which way to get rid of.

Reply:
i would prefer not using the "qualifier" - the code is still "segregated", and it's more explicit with a guard condition.